### PR TITLE
Add section and TOC nav for EngCom best practices

### PR DESCRIPTION
**Pull request by lorikrell**
## This PR is a:

- [X] New topic
- [ ] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

When this pull request is merged, it will...

Add navigation (TOC) to the Community Resources area to support a best practices guide (name subject to change). Added example commented-out content for potential children pages.
Move content from resources to new best practices parent page. Add link to resources to it.
Rename index.md to resources.md.
Add Slack info for resources. 
Add recommended presentation on caching/indexing to resources.
Update site index link to Community Resources and change order of Community links.
Typo fixes.
Remove an old best practices file and redirect to Performance files.

### CODE REVIEW

File: community/resources/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- [Store Configuration for Commerce Cloud]({{ page.baseurl }}/cloud/configure/configure-best-practices.html)
+- [Deployment for Commerce Cloud]({{ page.baseurl }}/cloud/reference/discover-deploy.html)
+
+## Want to contribute?
+
+You can [contribute](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md) by adding content and new topics through Pull Requests to the [DevDocs](https://github.com/magento/devdocs) repo. Save all files to the `/community/resources` directory.
+
+We can help add navigation and format content when reviewing your submitted PRs. 
+
+If you have any questions, contact us through [#DevDocs Slack](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md) (or [join us](https://t.co/9HImUyCmyh)) or Twitter [@MagentoDevDocs](https://twitter.com/MagentoDevDocs).
```
Comment (keharper) [inline community/resources/best-practices.md]: The DevDocs Slack link goes to the CONTRIBUTING.md file.

File: community/resources/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+This section covers a broad scope of Community contributed best and leading practices, common solutions, and recommendations for Magento. 
+
+Topics and areas include:
+
+* Project Setup
+* Theme development (when and were to make a new theme)
+* Composer use (when to create a new repo)
+* Local environment options
+
+<div class="bs-callout bs-callout-info" markdown="1">
```
Comment (keharper) [inline community/resources/best-practices.md]: Use the {:.bs-callout .bs-callout-info} structure instead a div

File: community/resources/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- [Best Practices for Store Configuration]({{ page.baseurl }}/cloud/configure/configure-best-practices.html) for Commerce Cloud
+- [Deployment process]({{ page.baseurl }}/cloud/reference/discover-deploy.html) for Commerce Cloud
+
+## Want to contribute?
+
+You can [contribute](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md) by adding content and new topics through Pull Requests to the [DevDocs](https://github.com/magento/devdocs) repo. Save all files to the `/community/resources` directory. 
+
+We can help add navigation and format content when reviewing your submitted PRs. 
+
+If you have any questions, contact us through [#DevDocs Slack](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md) (or [join us](https://t.co/9HImUyCmyh)) or Twitter [@MagentoDevDocs](https://twitter.com/MagentoDevDocs).
```
Comment (keharper) [inline community/resources/best-practices.md]: The DevDocs Slack link goes to CONTRIBUTING.md

File: guides/v2.2/performance-best-practices/index.md [hunk 4]
```diff
@@ -2,7 +2,9 @@
 group: performance-best-practices
 title: Performance Best Practices
 landing-page: Performance Best Practices
-redirect_from: /guides/v2.2/config-guide/prod/prod_perf-optimize.html
+redirect_from:
```
Comment (keharper) [inline guides/v2.2/performance-best-practices/index.md]: This is the file that should be renamed (moreso than the one you did). It's a top-level index file.

Also add the following line to the front matter:

permalink: performance-best-practices

Comment (keharper) [reply guides/v2.2/performance-best-practices/index.md]: Don't add the permalink line yet. This has to wait until we complete the directory restructuring project

File: community/resources/best-practices.md (hunk 2 above)
Comment (lorikrell) [reply community/resources/best-practices.md]: Fixed, also fixed in sources.md

File: community/resources/best-practices.md (hunk 3 above)
Comment (lorikrell) [reply community/resources/best-practices.md]: Oops thanks! fixing

File: guides/v2.2/performance-best-practices/index.md (hunk 4 above)
Comment (lorikrell) [reply guides/v2.2/performance-best-practices/index.md]: Fixed, renamed the file, added redirect, updated symlink for 2.3

File: community/resources/best-practices.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- [Theme development]({{ page.baseurl }}/frontend-dev-guide/theme-best-practice.html)
+- [Performance]({{ page.baseurl }}/performance-best-practices/)
+- [Store Configuration for Commerce Cloud]({{ page.baseurl }}/cloud/configure/configure-best-practices.html)
+- [Deployment for Commerce Cloud]({{ page.baseurl }}/cloud/reference/discover-deploy.html)
+
+## Want to contribute?
+
+You can [contribute](https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md) by adding content and new topics through Pull Requests to the [DevDocs](https://github.com/magento/devdocs) repo. Save all files to the `/community/resources` directory.
+
+We can help add navigation and format content when reviewing your submitted PRs.
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply community/resources/best-practices.md]: Added info: Some PRs may require architectural and internal development reviews to verify and approve the contributed best practices. These reviews may require more time to complete before merging content.

**Comment by lorikrell** [comment]
Pushed updates for all review comments! Thank you @keharper !

**Comment by dshevtsov** [comment]
@lorikrell please add `whatsnew`

**Comment by lorikrell** [comment]
@dshevtsov  Sorry for missing that, added!

**Pull request merged**